common/include/algebra/concepts.hpp

```diff
@@ -88,6 +88,41 @@ concept column_matrix = matrix<M> && (algebra::traits::columns<M> == 1);
 
 template <typename M>
 concept column_matrix3D = column_matrix<M> && (algebra::traits::rows<M> == 3);
+
+template <typename MA, typename MB>
+concept matrix_multipliable =
+    matrix<MA>&& matrix<MB> &&
+    (algebra::traits::columns<MA> ==
+     algebra::traits::rows<
+         MB>)&&std::convertible_to<algebra::traits::index_t<MA>,
```

Comment:
Should we maybe factor a part of this concept into a smaller concept like e.g. ```compatible_matrix```, where we test the index and scalar types?